dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/monitor/DatadogRumMonitor.kt

```diff
@@ -130,18 +134,39 @@ internal class DatadogRumMonitor(
             devLogger.w("User action $action was ignored: previous action still active.")
             return
         }
-        sendUnclosedAction()
-        val nanoTime = System.nanoTime()
-        val eventData = RumEventData.UserAction(
-            action,
-            UUID.randomUUID(),
-            nanoTime
-        )
+        registerUserAction(action, attributes)
+    }
 
-        val event = rumEvent(eventData, attributes)
-        lastActionRelatedEventNs.set(nanoTime)
-        activeActionEvent = event
-        activeActionData = eventData
+    @Synchronized
+    internal fun startUserAction() {
+
+        if (isWithinActionScope()) {
+            devLogger.w("User action was ignored: previous action still active.")
+            return
+        }
+
+        registerUserAction("", emptyMap())
+        // mark this action as waiting for being manually closed
+        isWaitingForManualClosing = true
+    }
+
+    @Synchronized
+    internal fun stopUserAction(action: String, attributes: Map<String, Any?>) {
```

Comment:
What happens if you stop the user action, but `isWaitingForManualClosing` is already false ?

Reply:
Good point....it should do nothing. I will fix this and add an unit test there